crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -1,80 +1,420 @@
+pub struct AdditionalData {
+    masked_pan: String,
+    card_id: Secret<String>,
+    card_id4: Option<Secret<String>>,
+    card_expiry_date: Option<Secret<String>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RedirectPayload {
+    #[serde(rename = "PaRes")]
+    pa_res: String,
+
+    #[serde(rename = "paymentId")]
+    payment_id: String,
+}
+
+impl TryFrom<&PaymentsPreProcessingRouterData> for NexixpayPreProcessingRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &PaymentsPreProcessingRouterData) -> Result<Self, Self::Error> {
+        let redirect_response = item.request.redirect_response.clone().ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "redirect_response",
+            },
+        )?;
+        let redirect_payload = redirect_response
+            .payload
+            .ok_or(errors::ConnectorError::MissingConnectorRedirectionPayload {
+                field_name: "request.redirect_response.payload",
+            })?
+            .expose();
+        let customer_details_encrypted: RedirectPayload =
+            serde_json::from_value::<RedirectPayload>(redirect_payload.clone()).change_context(
+                errors::ConnectorError::MissingConnectorRedirectionPayload {
+                    field_name: "redirection_payload",
+                },
+            )?;
+        Ok(Self {
+            operation_id: customer_details_encrypted.payment_id,
+            three_d_s_auth_response: customer_details_encrypted.pa_res,
+        })
+    }
+}
+
+impl<F>
+    TryFrom<
+        ResponseRouterData<
+            F,
+            NexixpayPreProcessingResponse,
+            PaymentsPreProcessingData,
+            PaymentsResponseData,
+        >,
+    > for RouterData<F, PaymentsPreProcessingData, PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<
+            F,
+            NexixpayPreProcessingResponse,
+            PaymentsPreProcessingData,
+            PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let three_ds_data = item.response.three_d_s_auth_result;
+        let customer_details_encrypted: RedirectPayload = item
+            .data
+            .request
+            .redirect_response
+            .as_ref()
+            .and_then(|res| res.payload.to_owned())
+            .ok_or(errors::ConnectorError::MissingConnectorRedirectionPayload {
+                field_name: "request.redirect_response.payload",
+            })?
+            .expose()
+            .parse_value("RedirectPayload")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let is_auto_capture = item.data.request.is_auto_capture()?;
+        let operation_id = item.response.operation.operation_id;
+        let connector_metadata = Some(serde_json::json!(NexixpayConnectorMetaData {
```

Comment:
Use UpdateConnectorMetaData here and pass other values as none, if we are not intended to update them.

Reply:
this would be against SRP as the function is intended to update metadata itself.